### PR TITLE
Move the renaming of llvm-generated __invoke_XX functions from s2wasm into wasm-emscripten

### PR DESCRIPTION
**Pull request by sbc100**
This allows the same functionality to be using also in
wasm-emscripten-finalize.

### CODE REVIEW

**Comment by sbc100** [comment]
Test with llvm backend everything passes in `./tests/runner.py binaryen2`

Also tested with `EMCC_EXPERIMENTAL_USE_LLD=1  ./tests/runner.py binaryen2` and the failures went from `FAILED (failures=6, errors=66)` to `FAILED (failures=3, errors=18)`

**Comment by sbc100** [comment]
This code does rely on the specific order of visito: visitImport -> visitTable -> visitCallImport -> visitModule.  Is that a reasonable assumption, or should I instead made several passes that run in sequence?  @kripken ?

**Comment by kripken** [comment]
Hmm, the current order according to wasm-traversal.cpp:337 is import > function contents (callImport) > table, which is different.

In general we didn't document a specific order, but it's possible passes depend on it...

File: src/wasm/wasm-emscripten.cpp [hunk 1]
```diff
@@ -508,6 +508,123 @@ void EmscriptenGlueGenerator::fixEmAsmConsts() {
...
+      // Mark this (now duplicate) import for deletion.
+      toRemove.push_back(curr->name);
+    } else {
+      // Rename this import to the new name
+      curr->base = newname;
+      curr->name = newname;
+    }
+  }
+
+  void visitCallImport(CallImport* curr) {
```
[the remark this answers is not in the document]
Comment (jgravelle-google) [reply src/wasm/wasm-emscripten.cpp]: Talked locally. This is unlikely to change, and is covered by the `test/dot_s/fix_em_ehsjlj_names.wast` test, so is unlikely to break without CI noticing. Lgtm

**Pull request merged**